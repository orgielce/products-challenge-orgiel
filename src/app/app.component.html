<router-outlet></router-outlet>

<ngx-spinner size="large" id="loader" type="timer">
  <div class="loader">
    <div class="mask mask1">
      <img class="color color1" src="assets/spinner/color1.png" alt="Color 1">
    </div>
    <div class="mask mask2">
      <img class="color color2" src="assets/spinner/color2.png" alt="Color 2">
    </div>
  </div>
</ngx-spinner>
