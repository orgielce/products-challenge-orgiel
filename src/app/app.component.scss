.loader {
  width: 300px;
  height: 90px;
  position: relative;
}
.loader .mask {
  width: 300px;
  height: 90px;
  position: absolute;
  left: 0;
  top: 0;
}
.color {
  position: absolute;
  bottom: 0;
  -webkit-animation: maskAnim 2s ease infinite;
  animation: maskAnim 2s ease infinite;
}
.color.color1,
.color.color2 {
  width: 0;
  left: 0;
  right: initial;
  height: 90px;
}
.mask1 {
  -webkit-mask-image: url(src/assets/spinner/mask4.svg);
  mask-image: url(src/assets/spinner/mask4.svg);
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
}
.mask2 {
  -webkit-mask-image: url(src/assets/spinner/mask3.svg);
  mask-image: url(src/assets/spinner/mask3.svg);
  -webkit-mask-repeat: no-repeat;
  mask-repeat: no-repeat;
}

@-webkit-keyframes maskAnim {
  5%   { width: 0; left: 0; right: initial }
  35%  { width: 100%; left: 0; right: initial }
  65%  { width: 100%; left: initial; right: 0 }
  95% { width: 0; left: initial; right: 0 }
}

@keyframes maskAnim {
  5%   { width: 0; left: 0; right: initial }
  35%  { width: 100%; left: 0; right: initial }
  65%  { width: 100%; left: initial; right: 0 }
  95% { width: 0; left: initial; right: 0 }
}

#loader {
  position: fixed;
  left: 0;
  top: 0;
  z-index: 10;
  height: 10px;
  background-color: transparent;
}

.ngx-spinner-overlay {
  top: 50% !important;
  left: 50% !important;
  margin: 0;
  position: absolute;
  transform: translate(-50%,-50%) !important;
  background-color: rgba(204, 205, 210, 0.78) !important;

  .loading-text {
    position: absolute;
    top: 50% !important;
    left: 50% !important;
    transform: translate(-50%,-50%) !important;
  }
}
