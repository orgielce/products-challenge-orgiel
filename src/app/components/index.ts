export * from './gallery';
export * from './header';
export * from './page-not-found';
export * from './paginator';
export * from './table';
