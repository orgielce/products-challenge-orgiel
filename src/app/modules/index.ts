
export * from './auth';
export * from './layout';
export * from './products';
export * from './store.module';
export * from './primeng.module';
