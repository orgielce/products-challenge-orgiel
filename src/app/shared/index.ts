
export * from './helpers';
export * from './models';
export * from './enums/routes';
export * from './services';
export * from './store';
export * from './guards';
export * from './enums';
export * from './filters';
