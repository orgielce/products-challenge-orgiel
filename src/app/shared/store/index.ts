export * from './app.states';
export * from './actions';
export * from './effects';
export * from './reducers';
