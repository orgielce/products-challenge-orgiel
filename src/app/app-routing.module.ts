import {NgModule} from '@angular/core';
import {RouterModule, Routes} from '@angular/router';
import {ROUTES_PATH} from "./shared";
import {PageNotFoundComponent} from "./components";

const routes: Routes = [
  {
    path: ROUTES_PATH.Base,
    loadChildren: () => import('./modules').then(m => m.LayoutModule)
  }, {
    path: ROUTES_PATH.Login,
    loadChildren: () => import('./modules').then(m => m.AuthModule)
  }, {
    path: '**', component: PageNotFoundComponent
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
