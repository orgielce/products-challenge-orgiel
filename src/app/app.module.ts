import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {BrowserAnimationsModule} from "@angular/platform-browser/animations";

import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './app.component';

import {HTTP_INTERCEPTORS, HttpClientModule} from "@angular/common/http";
import {KeyInterceptor} from "./shared";

import {NgRxModule} from "./modules/store.module";

import {NgxSpinnerModule} from "ngx-spinner";

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    AppRoutingModule,
    NgxSpinnerModule,
    // ngrx
    NgRxModule
  ],
  providers: [
    {
      provide: HTTP_INTERCEPTORS,
      useClass: KeyInterceptor,
      multi: true
    }
  ],
  bootstrap: [AppComponent]
})
export class AppModule {
}
