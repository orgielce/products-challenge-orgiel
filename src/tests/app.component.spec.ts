import {ComponentFixture, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {AppComponent} from '../app/app.component';
import {BrowserModule} from "@angular/platform-browser";
import {HttpClientModule} from "@angular/common/http";
import {NgRxModule} from "../app/modules/store.module";


describe('AppComponent', () => {

  let component: AppComponent;
  let fixture: ComponentFixture<AppComponent>;
  let compiled: HTMLElement;

  beforeEach(() => TestBed.configureTestingModule({
    imports: [
      RouterTestingModule,
      BrowserModule,
      // BrowserAnimationsModule,
      HttpClientModule,
      NgRxModule
    ],
    declarations: [AppComponent]
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    component = fixture.componentInstance;

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create the app', () => {
    expect(component).toBeTruthy();
  });

  test(`should have as title 'Challenge products by Orgiel'`, () => {
    expect(component.title).toEqual('Challenge products by Orgiel');
  });

  // test(`should found ngx-spinner'`, () => {
  //   const ngxSpinner = compiled.querySelector('ngx-spinner');
  //   expect( ngxSpinner ).toBeTruthy();
  // });

  test('should match with snapshot', () => {
    fixture.detectChanges();
    expect(compiled).toMatchSnapshot();
  });
});
