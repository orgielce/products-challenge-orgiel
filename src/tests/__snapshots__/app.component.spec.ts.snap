// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match with snapshot 1`] = `
<div
  id="root2"
>
  <router-outlet />
  
  <ngx-spinner
    id="loader"
    size="large"
    type="timer"
  >
    
  </ngx-spinner>
</div>
`;
